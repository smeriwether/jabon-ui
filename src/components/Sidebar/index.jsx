import React, { Component } from 'react';

import AccountModal from './components/AccountModal';
import Logo from '../../UI/Logo';
import Account from '../../UI/Account';

class Sidebar extends Component {
  constructor(props) {
    super(props);

    this.state = {
      openAccountModal: false
    };
  }

  toggleAccountModal = () => {
    this.setState((prevState) => ({
      ...prevState,
      openAccountModal: !prevState.openAccountModal
    }));
  }

  render() {
    return (
      <div className='h-100 w-100 pb3 bg-blue flex flex-column justify-between items-center'>
        <Logo fill='#357EDD' />
        <Account fill='white' onClick={ this.toggleAccountModal } />

        <AccountModal open={ this.state.openAccountModal } onClose={ this.toggleAccountModal } />
      </div>
    );
  }
};

export default Sidebar;
