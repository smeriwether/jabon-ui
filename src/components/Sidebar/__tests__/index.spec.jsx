import Sidebar from '../../Sidebar';

describe('<Sidebar />', () => {
  const setup = () => {
    const wrapper = shallow(<Sidebar />);
    return { wrapper };
  }

  it('should match the snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
