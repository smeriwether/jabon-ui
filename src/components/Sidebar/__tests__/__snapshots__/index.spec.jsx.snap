// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Logo
          fill="#357EDD"
        />,
        <Account
          fill="white"
          onClick={[Function]}
        />,
        <AccountModal
          onClose={[Function]}
          open={false}
        />,
      ],
      "className": "h-100 w-100 pb3 bg-blue flex flex-column justify-between items-center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "fill": "#357EDD",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "fill": "white",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClose": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Logo
            fill="#357EDD"
          />,
          <Account
            fill="white"
            onClick={[Function]}
          />,
          <AccountModal
            onClose={[Function]}
            open={false}
          />,
        ],
        "className": "h-100 w-100 pb3 bg-blue flex flex-column justify-between items-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "fill": "#357EDD",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "fill": "white",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClose": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
